### ISSUE TITLE
Half-page scrolling leaves some lines unchanged on macOS with TERM=tmux-256color

### ISSUE THREAD

**Issue by achilleas-k** [tmux/tmux#1584]
Half-page scrolling in some applications (weechat and tig) leaves lines unchanged with TERM=tmux-256color on macOS

Info:
- Tmux version: 2.8
- OS: macOS 10.14.2 (18C54) (not reproducible on Linux with the same setup)
- TERM outside tmux: `xterm-256color`
- TERM inside tmux: `tmux-256color` (issue does not occur with `TERM=screen` or `TERM=screen-256color`)
- Issue presents in both iTerm2 and Terminal.app
- Steps to reproduce the problem:
  1. Run weechat or tig inside a tmux session
  2. Switch to a scrollable view with full lines (long noisy buffer in weechat or long git history in tig)
  3. Scroll the view up by half a page (Alt+Pgup in weechat, Ctrl+U in tig)
  4. Several lines near the bottom of the view remain unchanged (see attached recordings)

- Refreshing the screen in either application (Ctrl+L) clears the issue
- The issue seems to only occur when scrolling up, not down
- Doing a full page scroll (page up) doesn't trigger the issue
- A single line configuration file with just `set -g default-terminal "tmux-256color"` is enough to cause the issue

```
❯ uname -sp && tmux -V && echo $TERM
Darwin i386
tmux 2.8
tmux-256color
```

The logs and screen recordings were created with the single line tmux.conf that sets the TERM to `tmux-256color`, using version 2.8.
Given the nature of the bug, if it's hard to follow what's happening on screen in terms of scrolling, the status line of tig shows the highlighted commit number and makes it easier to figure out when the screen is scrolling up or down, half or full page.

The issue is reproducible on current master HEAD.

Attachments and links:
- Server logs: [tmux-server-logs.zip](https://github.com/tmux/tmux/files/2776759/tmux-server-logs.zip)
- Screen recordings (YouTube): [tig](https://youtu.be/JXcr_zbD2f8) and [weechat](https://youtu.be/sMPPYoRYANU)

**Comment by nicm** [comment]
Sounds like a terminal bug to me, did you try asking iterm2 developers?

**Comment by nicm** [comment]
Does doing this in .tmux.conf make any difference:
~~~~
set -as terminal-overrides ',*:indn@'

**Comment by achilleas-k** [comment]
I was convinced it was an iTerm2 issue and while writing up the report for that, I reproduced it in Terminal.app as well.

**Comment by nicm** [comment]
Terminal.app is full of bugs so I probably wouldn't worry about it, iterm2 is better though. But you can't reproduce on Linux with any terminal?

**Comment by achilleas-k** [comment]
> But you can't reproduce on Linux with any terminal?
I haven't actively tried to reproduce on Linux, but I use the same configuration in Konsole and Termite and it doesn't occur.

Speaking of terminals, I just tried it now with Alacritty on macOS and it still occurs.
`set -as terminal-overrides ',*:indn@'` doesn't fix the issue.

**Comment by nicm** [comment]
Can you show me two screenshots, one before scrolling and one after, together with the corresponding server log?

**Comment by achilleas-k** [comment]
Here you go. Configuration was
```
set -g default-terminal "tmux-256color"
set -gas terminal-overrides ',*:indn@'
```
[tmux-server-98375.log](https://github.com/tmux/tmux/files/2776784/tmux-server-98375.log)

<img width="1102" alt="before scroll" src="https://user-images.githubusercontent.com/2369197/51441694-4226e300-1cd4-11e9-8565-5f1b93efee63.png">


<img width="1102" alt="after scroll" src="https://user-images.githubusercontent.com/2369197/51441693-4226e300-1cd4-11e9-9d22-1795d56b4101.png">

**Comment by nicm** [comment]
Which lines are unchanged here?

**Comment by nicm** [comment]
Does pressing C-b r rather than C-l fix it?

**Comment by nicm** [comment]
Can you show me the output of "infocmp -x tmux-256color?

This application appears to be sending \033[T which is SD (Scroll Down) and is not supported by tmux, and is not present in the standard screen-256color or tmux-256color as far as I can see.

**Comment by nicm** [comment]
Also "infocmp -x tmux-256color screen-256color" would be nice as well.

**Comment by achilleas-k** [comment]
```
#	Reconstructed via infocmp from file: /Users/achilleas/.terminfo/74/tmux-256color
tmux-256color|tmux with 256 colors,
	OTbs, OTpt, am, bce, ccc, km, mc5i, mir, msgr, npc, xenl, AX, G0,
	colors#256, cols#80, it#8, lines#24, ncv#3, pairs#32767,
	acsc=``aaffggiijjkkllmmnnooppqqrrssttuuvvwwxxyyzz{{||}}~~,
	bel=^G, blink=\E[5m, bold=\E[1m, cbt=\E[Z, civis=\E[?25l,
	clear=\E[H\E[2J, cnorm=\E[?12l\E[?25h, cr=^M,
	csr=\E[%i%p1%d;%p2%dr, cub=\E[%p1%dD, cub1=^H,
	cud=\E[%p1%dB, cud1=^J, cuf=\E[%p1%dC, cuf1=\E[C,
	cup=\E[%i%p1%d;%p2%dH, cuu=\E[%p1%dA, cuu1=\E[A,
	cvvis=\E[?12;25h, dch=\E[%p1%dP, dch1=\E[P, dl=\E[%p1%dM,
	dl1=\E[M, ech=\E[%p1%dX, ed=\E[J, el=\E[K, el1=\E[1K,
	enacs=\E(B\E)0, flash=\E[?5h$<100/>\E[?5l, home=\E[H,
	hpa=\E[%i%p1%dG, ht=^I, hts=\EH, ich=\E[%p1%d@,
	il=\E[%p1%dL, il1=\E[L, ind=^J, indn=\E[%p1%dS,
	initc=\E]4;%p1%d;rgb\:%p2%{255}%*%{1000}%/%2.2X/%p3%{255}%*%{1000}%/%2.2X/%p4%{255}%*%{1000}%/%2.2X\E\\,
	invis=\E[8m, is2=\E[!p\E[?3;4l\E[4l\E>, kDC=\E[3;2~,
	kEND=\E[1;2F, kHOM=\E[1;2H, kIC=\E[2;2~, kLFT=\E[1;2D,
	kNXT=\E[6;2~, kPRV=\E[5;2~, kRIT=\E[1;2C, kb2=\EOE, kbs=^H,
	kcbt=\E[Z, kcub1=\EOD, kcud1=\EOB, kcuf1=\EOC, kcuu1=\EOA,
	kdch1=\E[3~, kend=\E[4~, kent=\EOM, kf1=\EOP, kf10=\E[21~,
	kf11=\E[23~, kf12=\E[24~, kf13=\E[1;2P, kf14=\E[1;2Q,
	kf15=\E[1;2R, kf16=\E[1;2S, kf17=\E[15;2~, kf18=\E[17;2~,
	kf19=\E[18;2~, kf2=\EOQ, kf20=\E[19;2~, kf21=\E[20;2~,
	kf22=\E[21;2~, kf23=\E[23;2~, kf24=\E[24;2~,
	kf25=\E[1;5P, kf26=\E[1;5Q, kf27=\E[1;5R, kf28=\E[1;5S,
	kf29=\E[15;5~, kf3=\EOR, kf30=\E[17;5~, kf31=\E[18;5~,
	kf32=\E[19;5~, kf33=\E[20;5~, kf34=\E[21;5~,
	kf35=\E[23;5~, kf36=\E[24;5~, kf37=\E[1;6P, kf38=\E[1;6Q,
	kf39=\E[1;6R, kf4=\EOS, kf40=\E[1;6S, kf41=\E[15;6~,
	kf42=\E[17;6~, kf43=\E[18;6~, kf44=\E[19;6~,
	kf45=\E[20;6~, kf46=\E[21;6~, kf47=\E[23;6~,
	kf48=\E[24;6~, kf49=\E[1;3P, kf5=\E[15~, kf50=\E[1;3Q,
	kf51=\E[1;3R, kf52=\E[1;3S, kf53=\E[15;3~, kf54=\E[17;3~,
	kf55=\E[18;3~, kf56=\E[19;3~, kf57=\E[20;3~,
	kf58=\E[21;3~, kf59=\E[23;3~, kf6=\E[17~, kf60=\E[24;3~,
	kf61=\E[1;4P, kf62=\E[1;4Q, kf63=\E[1;4R, kf7=\E[18~,
	kf8=\E[19~, kf9=\E[20~, khome=\E[1~, kich1=\E[2~,
	kind=\E[1;2B, kmous=\E[M, knp=\E[6~, kpp=\E[5~,
	kri=\E[1;2A, mc0=\E[i, mc4=\E[4i, mc5=\E[5i, meml=\El,
	memu=\Em, nel=\EE, op=\E[39;49m, rc=\E8, rev=\E[7m, ri=\EM,
	rin=\E[%p1%dT, ritm=\E[23m, rmacs=\E(B, rmam=\E[?7l,
	rmcup=\E[?1049l, rmir=\E[4l, rmkx=\E[?1l\E>,
	rmm=\E[?1034l, rmso=\E[27m, rmul=\E[24m, rs1=\Ec,
	rs2=\E[!p\E[?3;4l\E[4l\E>, sc=\E7,
	setab=\E[%?%p1%{8}%<%t4%p1%d%e%p1%{16}%<%t10%p1%{8}%-%d%e48;5;%p1%d%;m,
	setaf=\E[%?%p1%{8}%<%t3%p1%d%e%p1%{16}%<%t9%p1%{8}%-%d%e38;5;%p1%d%;m,
	sgr=%?%p9%t\E(0%e\E(B%;\E[0%?%p6%t;1%;%?%p2%t;4%;%?%p1%p3%|%t;7%;%?%p4%t;5%;%?%p7%t;8%;m,
	sgr0=\E(B\E[m, sitm=\E[3m, smacs=\E(0, smam=\E[?7h,
	smcup=\E[?1049h, smir=\E[4h, smkx=\E[?1h\E=,
	smm=\E[?1034h, smso=\E[7m, smul=\E[4m, tbc=\E[3g,
	u6=\E[%i%d;%dR, u7=\E[6n, u8=\E[?1;2c, u9=\E[c,
	vpa=\E[%i%p1%dd, E0=\E(B, Ms@, S0=\E(%p1%c, kDC3=\E[3;3~,
	kDC4=\E[3;4~, kDC5=\E[3;5~, kDC6=\E[3;6~, kDC7=\E[3;7~,
	kDN=\E[1;2B, kDN3=\E[1;3B, kDN4=\E[1;4B, kDN5=\E[1;5B,
	kDN6=\E[1;6B, kDN7=\E[1;7B, kEND3=\E[1;3F, kEND4=\E[1;4F,
	kEND5=\E[1;5F, kEND6=\E[1;6F, kEND7=\E[1;7F,
	kHOM3=\E[1;3H, kHOM4=\E[1;4H, kHOM5=\E[1;5H,
	kHOM6=\E[1;6H, kHOM7=\E[1;7H, kIC3=\E[2;3~, kIC4=\E[2;4~,
	kIC5=\E[2;5~, kIC6=\E[2;6~, kIC7=\E[2;7~, kLFT3=\E[1;3D,
	kLFT4=\E[1;4D, kLFT5=\E[1;5D, kLFT6=\E[1;6D,
	kLFT7=\E[1;7D, kNXT3=\E[6;3~, kNXT4=\E[6;4~,
	kNXT5=\E[6;5~, kNXT6=\E[6;6~, kNXT7=\E[6;7~,
	kPRV3=\E[5;3~, kPRV4=\E[5;4~, kPRV5=\E[5;5~,
	kPRV6=\E[5;6~, kPRV7=\E[5;7~, kRIT3=\E[1;3C,
	kRIT4=\E[1;4C, kRIT5=\E[1;5C, kRIT6=\E[1;6C,
	kRIT7=\E[1;7C, kUP=\E[1;2A, kUP3=\E[1;3A, kUP4=\E[1;4A,
	kUP5=\E[1;5A, kUP6=\E[1;6A, kUP7=\E[1;7A,
```

**Comment by nicm** [comment]
This is not supported by tmux and should not be present:
~~~~
rin=\E[%p1%dT
~~~~

**Comment by nicm** [comment]
Are you using the system tmux-256color or did you build your own?

**Comment by achilleas-k** [comment]
> Are you using the system tmux-256color or did you build your own?

Oh yes. I did make a modification some time ago based and completely forgot about that. This is rather embarrassing, sorry.

Let me restore the system terminfo and check again.

**Comment by achilleas-k** [comment]
Oh now I remember. The reason it's not the system one is because there is no tmux-256color terminfo by default.

**Comment by nicm** [comment]
It looks like you copied it from xterm which is wrong. It should look like this:
~~~~
#	Reconstructed via infocmp from file: /home/nicholas/.terminfo/t/tmux-256color
tmux-256color|tmux with 256 colors,
	OTbs, OTpt, am, hs, km, mir, msgr, xenl, AX, G0,
	colors#0x100, cols#80, it#8, lines#24, pairs#0x7fff, U8#1,
	acsc=++\,\,--..00``aaffgghhiijjkkllmmnnooppqqrrssttuuvvwwxxyyzz{{||}}~~,
	bel=^G, blink=\E[5m, bold=\E[1m, cbt=\E[Z, civis=\E[?25l,
	clear=\E[H\E[J, cnorm=\E[34h\E[?25h, cr=\r,
	csr=\E[%i%p1%d;%p2%dr, cub=\E[%p1%dD, cub1=^H,
	cud=\E[%p1%dB, cud1=\n, cuf=\E[%p1%dC, cuf1=\E[C,
	cup=\E[%i%p1%d;%p2%dH, cuu=\E[%p1%dA, cuu1=\EM,
	cvvis=\E[34l, dch=\E[%p1%dP, dch1=\E[P, dim=\E[2m,
	dl=\E[%p1%dM, dl1=\E[M, dsl=\E]0;\007, ed=\E[J, el=\E[K,
	el1=\E[1K, enacs=\E(B\E)0, flash=\Eg, fsl=^G, home=\E[H,
	hpa=\E[%i%p1%dG, ht=^I, hts=\EH, ich=\E[%p1%d@,
	il=\E[%p1%dL, il1=\E[L, ind=\n, indn=\E[%p1%dS,
	invis=\E[8m, is2=\E)0, kDC=\E[3;2~, kEND=\E[1;2F,
	kHOM=\E[1;2H, kIC=\E[2;2~, kLFT=\E[1;2D, kNXT=\E[6;2~,
	kPRV=\E[5;2~, kRIT=\E[1;2C, kbs=^H, kcbt=\E[Z, kcub1=\EOD,
	kcud1=\EOB, kcuf1=\EOC, kcuu1=\EOA, kdch1=\E[3~,
	kend=\E[4~, kf1=\EOP, kf10=\E[21~, kf11=\E[23~,
	kf12=\E[24~, kf13=\E[1;2P, kf14=\E[1;2Q, kf15=\E[1;2R,
	kf16=\E[1;2S, kf17=\E[15;2~, kf18=\E[17;2~,
	kf19=\E[18;2~, kf2=\EOQ, kf20=\E[19;2~, kf21=\E[20;2~,
	kf22=\E[21;2~, kf23=\E[23;2~, kf24=\E[24;2~,
	kf25=\E[1;5P, kf26=\E[1;5Q, kf27=\E[1;5R, kf28=\E[1;5S,
	kf29=\E[15;5~, kf3=\EOR, kf30=\E[17;5~, kf31=\E[18;5~,
	kf32=\E[19;5~, kf33=\E[20;5~, kf34=\E[21;5~,
	kf35=\E[23;5~, kf36=\E[24;5~, kf37=\E[1;6P, kf38=\E[1;6Q,
	kf39=\E[1;6R, kf4=\EOS, kf40=\E[1;6S, kf41=\E[15;6~,
	kf42=\E[17;6~, kf43=\E[18;6~, kf44=\E[19;6~,
	kf45=\E[20;6~, kf46=\E[21;6~, kf47=\E[23;6~,
	kf48=\E[24;6~, kf49=\E[1;3P, kf5=\E[15~, kf50=\E[1;3Q,
	kf51=\E[1;3R, kf52=\E[1;3S, kf53=\E[15;3~, kf54=\E[17;3~,
	kf55=\E[18;3~, kf56=\E[19;3~, kf57=\E[20;3~,
	kf58=\E[21;3~, kf59=\E[23;3~, kf6=\E[17~, kf60=\E[24;3~,
	kf61=\E[1;4P, kf62=\E[1;4Q, kf63=\E[1;4R, kf7=\E[18~,
	kf8=\E[19~, kf9=\E[20~, khome=\E[1~, kich1=\E[2~,
	kind=\E[1;2B, kmous=\E[M, knp=\E[6~, kpp=\E[5~,
	kri=\E[1;2A, nel=\EE, op=\E[39;49m, rc=\E8, rev=\E[7m,
	ri=\EM, ritm=\E[23m, rmacs=^O, rmcup=\E[?1049l, rmir=\E[4l,
	rmkx=\E[?1l\E>, rmso=\E[27m, rmul=\E[24m,
	rs2=\Ec\E[?1000l\E[?25h, sc=\E7,
	setab=\E[%?%p1%{8}%<%t4%p1%d%e%p1%{16}%<%t10%p1%{8}%-%d%e48;5;%p1%d%;m,
	setaf=\E[%?%p1%{8}%<%t3%p1%d%e%p1%{16}%<%t9%p1%{8}%-%d%e38;5;%p1%d%;m,
	sgr=\E[0%?%p6%t;1%;%?%p2%t;4%;%?%p1%p3%|%t;7%;%?%p4%t;5%;%?%p5%t;2%;%?%p7%t;8%;m%?%p9%t\016%e\017%;,
	sgr0=\E[m\017, sitm=\E[3m, smacs=^N, smcup=\E[?1049h,
	smir=\E[4h, smkx=\E[?1h\E=, smso=\E[7m, smul=\E[4m,
	tbc=\E[3g, tsl=\E]0;, vpa=\E[%i%p1%dd, Cr=\E]112\007,
	Cs=\E]12;%p1%s\007, E0=\E(B, E3=\E[3J,
	Ms=\E]52;%p1%s;%p2%s\007, S0=\E(%p1%c, Se=\E[2 q,
	Smulx=\E[4\:%p1%dm, Ss=\E[%p1%d q, TS=\E]0;,
	kDC3=\E[3;3~, kDC4=\E[3;4~, kDC5=\E[3;5~, kDC6=\E[3;6~,
	kDC7=\E[3;7~, kDN=\E[1;2B, kDN3=\E[1;3B, kDN4=\E[1;4B,
	kDN5=\E[1;5B, kDN6=\E[1;6B, kDN7=\E[1;7B, kEND3=\E[1;3F,
	kEND4=\E[1;4F, kEND5=\E[1;5F, kEND6=\E[1;6F,
	kEND7=\E[1;7F, kHOM3=\E[1;3H, kHOM4=\E[1;4H,
	kHOM5=\E[1;5H, kHOM6=\E[1;6H, kHOM7=\E[1;7H,
	kIC3=\E[2;3~, kIC4=\E[2;4~, kIC5=\E[2;5~, kIC6=\E[2;6~,
	kIC7=\E[2;7~, kLFT3=\E[1;3D, kLFT4=\E[1;4D,
	kLFT5=\E[1;5D, kLFT6=\E[1;6D, kLFT7=\E[1;7D,
	kNXT3=\E[6;3~, kNXT4=\E[6;4~, kNXT5=\E[6;5~,
	kNXT6=\E[6;6~, kNXT7=\E[6;7~, kPRV3=\E[5;3~,
	kPRV4=\E[5;4~, kPRV5=\E[5;5~, kPRV6=\E[5;6~,
	kPRV7=\E[5;7~, kRIT3=\E[1;3C, kRIT4=\E[1;4C,
	kRIT5=\E[1;5C, kRIT6=\E[1;6C, kRIT7=\E[1;7C, kUP=\E[1;2A,
	kUP3=\E[1;3A, kUP4=\E[1;4A, kUP5=\E[1;5A, kUP6=\E[1;6A,
	kUP7=\E[1;7A, rmxx=\E[29m, smxx=\E[9m,

**Comment by achilleas-k** [comment]
With that configuration, tig and weechat (and ranger) run without any colour at all. Note that these are all ncurses-based programs.

Using the one from this issue comment fixes the scrolling issue. https://github.com/tmux/tmux/issues/1257#issuecomment-365774651

I think I ended up with the problematic configuration while trying to get _italics_ to work, which isn't supported by the linked terminfo.

**Comment by achilleas-k** [comment]
But it's clear that the cause here is a broken/unsupported terminfo, so I think we can close the issue. Sorry for taking up your time with this.

**Comment by nicm** [comment]
Your ncurses might be too old, try this one. They both definitely support colour and this one does have italics however.
~~~~
#	Reconstructed via infocmp from file: /usr/share/terminfo/t/tmux-256color
tmux-256color|tmux with 256 colors,
	OTbs, OTpt, am, hs, km, mir, msgr, xenl, AX, G0,
	colors#256, cols#80, it#8, lines#24, pairs#32767, U8#1,
	acsc=++\,\,--..00``aaffgghhiijjkkllmmnnooppqqrrssttuuvvwwxxyyzz{{||}}~~,
	bel=^G, blink=\E[5m, bold=\E[1m, cbt=\E[Z, civis=\E[?25l,
	clear=\E[H\E[J, cnorm=\E[34h\E[?25h, cr=^M,
	csr=\E[%i%p1%d;%p2%dr, cub=\E[%p1%dD, cub1=^H,
	cud=\E[%p1%dB, cud1=^J, cuf=\E[%p1%dC, cuf1=\E[C,
	cup=\E[%i%p1%d;%p2%dH, cuu=\E[%p1%dA, cuu1=\EM,
	cvvis=\E[34l, dch=\E[%p1%dP, dch1=\E[P, dim=\E[2m,
	dl=\E[%p1%dM, dl1=\E[M, dsl=\E]0;\007, ed=\E[J, el=\E[K,
	el1=\E[1K, enacs=\E(B\E)0, flash=\Eg, fsl=^G, home=\E[H,
	ht=^I, hts=\EH, ich=\E[%p1%d@, il=\E[%p1%dL, il1=\E[L,
	ind=^J, is2=\E)0, kDC=\E[3;2~, kEND=\E[1;2F, kHOM=\E[1;2H,
	kIC=\E[2;2~, kLFT=\E[1;2D, kNXT=\E[6;2~, kPRV=\E[5;2~,
	kRIT=\E[1;2C, kbs=\177, kcbt=\E[Z, kcub1=\EOD, kcud1=\EOB,
	kcuf1=\EOC, kcuu1=\EOA, kdch1=\E[3~, kend=\E[4~, kf1=\EOP,
	kf10=\E[21~, kf11=\E[23~, kf12=\E[24~, kf13=\E[1;2P,
	kf14=\E[1;2Q, kf15=\E[1;2R, kf16=\E[1;2S, kf17=\E[15;2~,
	kf18=\E[17;2~, kf19=\E[18;2~, kf2=\EOQ, kf20=\E[19;2~,
	kf21=\E[20;2~, kf22=\E[21;2~, kf23=\E[23;2~,
	kf24=\E[24;2~, kf25=\E[1;5P, kf26=\E[1;5Q, kf27=\E[1;5R,
	kf28=\E[1;5S, kf29=\E[15;5~, kf3=\EOR, kf30=\E[17;5~,
	kf31=\E[18;5~, kf32=\E[19;5~, kf33=\E[20;5~,
	kf34=\E[21;5~, kf35=\E[23;5~, kf36=\E[24;5~,
	kf37=\E[1;6P, kf38=\E[1;6Q, kf39=\E[1;6R, kf4=\EOS,
	kf40=\E[1;6S, kf41=\E[15;6~, kf42=\E[17;6~,
	kf43=\E[18;6~, kf44=\E[19;6~, kf45=\E[20;6~,
	kf46=\E[21;6~, kf47=\E[23;6~, kf48=\E[24;6~,
	kf49=\E[1;3P, kf5=\E[15~, kf50=\E[1;3Q, kf51=\E[1;3R,
	kf52=\E[1;3S, kf53=\E[15;3~, kf54=\E[17;3~,
	kf55=\E[18;3~, kf56=\E[19;3~, kf57=\E[20;3~,
	kf58=\E[21;3~, kf59=\E[23;3~, kf6=\E[17~, kf60=\E[24;3~,
	kf61=\E[1;4P, kf62=\E[1;4Q, kf63=\E[1;4R, kf7=\E[18~,
	kf8=\E[19~, kf9=\E[20~, khome=\E[1~, kich1=\E[2~,
	kind=\E[1;2B, kmous=\E[M, knp=\E[6~, kpp=\E[5~,
	kri=\E[1;2A, nel=\EE, op=\E[39;49m, rc=\E8, rev=\E[7m,
	ri=\EM, ritm=\E[23m, rmacs=^O, rmcup=\E[?1049l, rmir=\E[4l,
	rmkx=\E[?1l\E>, rmso=\E[27m, rmul=\E[24m,
	rs2=\Ec\E[?1000l\E[?25h, sc=\E7,
	setab=\E[%?%p1%{8}%<%t4%p1%d%e%p1%{16}%<%t10%p1%{8}%-%d%e48;5;%p1%d%;m,
	setaf=\E[%?%p1%{8}%<%t3%p1%d%e%p1%{16}%<%t9%p1%{8}%-%d%e38;5;%p1%d%;m,
	sgr=\E[0%?%p6%t;1%;%?%p1%t;3%;%?%p2%t;4%;%?%p3%t;7%;%?%p4%t;5%;%?%p5%t;2%;m%?%p9%t\016%e\017%;,
	sgr0=\E[m\017, sitm=\E[3m, smacs=^N, smcup=\E[?1049h,
	smir=\E[4h, smkx=\E[?1h\E=, smso=\E[7m, smul=\E[4m,
	tbc=\E[3g, tsl=\E]0;, Cr=\E]112\007, Cs=\E]12;%p1%s\007,
	E0=\E(B, Ms=\E]52;%p1%s;%p2%s\007, S0=\E(%p1%c,
	Se=\E[2 q, Ss=\E[%p1%d q, TS=\E]0;, kDC3=\E[3;3~,
	kDC4=\E[3;4~, kDC5=\E[3;5~, kDC6=\E[3;6~, kDC7=\E[3;7~,
	kDN=\E[1;2B, kDN3=\E[1;3B, kDN4=\E[1;4B, kDN5=\E[1;5B,
	kDN6=\E[1;6B, kDN7=\E[1;7B, kEND3=\E[1;3F, kEND4=\E[1;4F,
	kEND5=\E[1;5F, kEND6=\E[1;6F, kEND7=\E[1;7F,
	kHOM3=\E[1;3H, kHOM4=\E[1;4H, kHOM5=\E[1;5H,
	kHOM6=\E[1;6H, kHOM7=\E[1;7H, kIC3=\E[2;3~, kIC4=\E[2;4~,
	kIC5=\E[2;5~, kIC6=\E[2;6~, kIC7=\E[2;7~, kLFT3=\E[1;3D,
	kLFT4=\E[1;4D, kLFT5=\E[1;5D, kLFT6=\E[1;6D,
	kLFT7=\E[1;7D, kNXT3=\E[6;3~, kNXT4=\E[6;4~,
	kNXT5=\E[6;5~, kNXT6=\E[6;6~, kNXT7=\E[6;7~,
	kPRV3=\E[5;3~, kPRV4=\E[5;4~, kPRV5=\E[5;5~,
	kPRV6=\E[5;6~, kPRV7=\E[5;7~, kRIT3=\E[1;3C,
	kRIT4=\E[1;4C, kRIT5=\E[1;5C, kRIT6=\E[1;6C,
	kRIT7=\E[1;7C, kUP=\E[1;2A, kUP3=\E[1;3A, kUP4=\E[1;4A,
	kUP5=\E[1;5A, kUP6=\E[1;6A, kUP7=\E[1;7A, rmxx=\E[29m,
	smxx=\E[9m,

**Comment by nicm** [comment]
Don't forget -x to tic.

**Comment by achilleas-k** [comment]
> Don't forget -x to tic.

That works (no italics though). Thanks again.

**Comment by nicm** [comment]
Did you set default-terminal?

**Comment by achilleas-k** [comment]
As you suggested earlier, the real issue is the old version of ncurses which is the system default on macOS (5.4). I have 6.1 installed via homebrew, but of course its terminfo database isn't used since it's only there to be linked to programs that are built against it (via homebrew).

The systemwide xterm-256color terminfo doesn't support italics.

Pointing the terminfo database to `/usr/local/Cellar/ncurses/6.1/share/terminfo` doesn't work (not sure why), so instead I dumped xterm-256color from a Linux machine with ncurses 6.1 and compiled it on macOS. So now my local `${HOME}/.terminfo` includes the tmux-256color you linked above and a xterm-256color from Linux and everything seems okay.

Is there any issue I might run into with the copying from Linux?

Thanks again.

**Comment by laggardkernel** [comment]
@achilleas-k I encountered the same problem. Terminfo from homebrew installed ncurses could not be used by tmux. Is it a bug of tmux?

**Comment by achilleas-k** [comment]
I didn't manage to get the homebrew ncurses terminfo database to work.  Couldn't dump from it, couldn't set the db path to it.  I don't know what I was doing wrong.

**Comment by laggardkernel** [comment]
@achilleas-k Homebrew installed ncurses terminfo database exists and could be dumped. The only problem is that it is not used by tmux.

Dump the terminfo from homebrew ncurses is easy.


```bash
# delete existing tmux-256color
rm -f ~/.terminfo/74/tmux-256color

# dump tmux-256color terminfo with brew-installed infocmp
/usr/local/opt/ncurses/bin/infocmp tmux-256color >| ./tmux-256color.terminfo
#	Reconstructed via infocmp from file: /usr/local/Cellar/ncurses/6.1/share/terminfo/74/tmux-256color
tmux-256color|tmux with 256 colors,
...

# compile it with pre-installed tic
/usr/bin/tic -x -o ~/.terminfo ./tmux-256color.terminfo
```

The problem is
1. terminfo file `/usr/local/Cellar/ncurses/6.1/share/terminfo/74/tmux-256color` could not used by tmux.
2. `/usr/local/Cellar/ncurses/6.1/share/terminfo/74/tmux-256color` is compiled as a newer version. We should recompile it with the old `/usr/bin/tic`.

**Comment by Darkshadow2** [comment]
You need to use `-x` with `infocmp` as well, or you're not getting the full definition of it.  But yes, you need to dump it with the `infocmp` from homebrew  and recompile it with the system `tic`, because the format of the compiled terminfo files changed between the two versions.  This is also why you can't just point `tmux` to the homebrew terminfo database.

Optionally, if you're compiling `tmux` yourself, you can set it to use homebrew's ncurses rather than the system one.

**Comment by laggardkernel** [comment]
@Darkshadow2 Did you mean tmux would use terminfo from `/usr/local/opt/ncurses/share/terminfo/` if it's compiled with ncurses installed by homebrew? And how to do it, add `depends_on "ncurses"` into `tmux.rb`?

**Comment by Darkshadow2** [comment]
@laggardkernel Yes, if it's compiled with ncurses installed by homebrew, it'll use that terminfo database.

I can't give specifics about how to do it, as I'm a caveman and compile everything manually.
